pom.xml

```diff
@@ -19,7 +19,7 @@
     </modules>
 
     <properties>
-        <guava.version>25.0-jre</guava.version>
+        <guava.version>29.0-jre</guava.version>
```

Comment:
This property is defined in a parent Maven file that is shared by every module in the project. Rewrite is able to look beyond just the XML content of this single file, understanding the relationship between this file and the Maven files in all of the repository's sub-modules. The upgrade version rule determines that this property is being used as a placholder in the [server module](https://github.com/openrewrite/sample-maven-dependencies/blob/master/server/pom.xml#L31).

Tools like Github's Dependabot would replace the version with `29.0-jre` in the server module, effectively breaking the association that project has with its parent going forward. Rewrite makes the connection between the two, leaves the property placeholder in place in the server module, and updates the property value here in the parent instead. This preserves the relationship going forward.